### ISSUE TITLE
Import/export false positive between namespace and variable

### ISSUE THREAD

**Issue by unional** [benmosher/eslint-plugin-import#2901]
```ts
export namespace testType { ... }

export const testType = { ... }
```

Results in: `error  Multiple exports of name 'testType'  import/export`

Here is an example: https://github.com/unional/type-plus/blob/main/type-plus/ts/testing/test_type.ts

**Comment by ljharb** [comment]
That still seems like a problem - which one does `import { testType }` point to?

**Comment by unional** [comment]
Namespace by itself does not produce a value. So there is not conflict and should work together. 

Types, values, and namespaces are 3 different structures and they can co-exists:

https://www.typescriptlang.org/docs/handbook/declaration-files/deep-dive.html

**Comment by ljharb** [comment]
right - but `import` can import either a type or a value, so which one would it be importing in this case?

**Comment by ljharb** [comment]
I do see that the docs explicitly say you can export a type and a value with the same name, so that's something we should technically allow, despite it seeming like a horrifically bad pattern for readability and clarity.

**Comment by unional** [comment]
It's actually a pretty good way to organize the code IMO.

So that instead of:

```ts
export type FooParam1 = {}
export type FooParam2 = {}
export type FooResult = { complex1: FooComplex, complex2: FooComplex2 }
export type FooComplex1 = {}
export type FooComplex2 = {}

export function foo(a: FooParam1, b: FooParam2): FooResult { ... }
```

You can do it like:

```ts
export namespace foo {
  export type Param1 = {}
  export type Param2 = {}
  export type Result = { complex1: Complex1, complex2: Complex2 }
  export type Complex1 = {}
  export type Complex2 = {}
}

export function foo(a: foo.Param1, b: foo.Param2): foo.Result { ... }
```

So that during consumption, you won't load of all those extra types:

```ts
import { foo, FooParams1, ... } from './foo.js'

export function wrapper(a: FooParam1) {
  return foo(a)
}

// vs
import { foo } from './foo.js'

export function wrapper(a: foo.Param1) {
  return foo(a)
}
```

Think of it as a way to "tuck" the extra types under the object/function.
